src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaMessageConsumer.cs

```diff
@@ -151,6 +150,45 @@ async Task ActiveAndActualAgentsCompleted(StopContext context)
                     LogContext.Warning?.Log("Stop canceled waiting for message consumers to complete: {InputAddress}", _context.InputAddress);
                 }
             }
+
+            await _task.ConfigureAwait(false);
+            await _executorPool.DisposeAsync().ConfigureAwait(false);
+
+            _consumer.Dispose();
+            _cancellationTokenSource.Dispose();
+        }
+
+
+        class CombinedChannelExecutorPool :
```

Comment:
Won't this be common across both Azure Event Hub and Kafka?

Reply:
I do not know if it worth extracting into common as it Ridier heavy, maybe?